### DISCUSSION TITLE
Edit Profile Cancel Button action

### DISCUSSION THREAD

**Discussion by jsalgado10** [OrchardCMS/OrchardCore#8411] [Q&A]
I am using the latest dev build 1.0.0-rc2-15787 and I created a viewer role that all they can do is view content, Access admin panel and Manage own user information. The only reason why the user has access to admin is so they can update their own profile. This is working, but when the user hits cancel they get a 403 error. It looks like the cancel button is redirecting to the Index action on the  Admin controller. 

is there anyway to overwrite the UserSaveButtons.Edit.cshtml view? 

I think the cancel button should redirect back the home page of the admin dashboard

**Comment by deanmarcussen** [comment]
Place a `UserSaveButtons.Edit.cshtml` in a module (not a theme), and take a dependency in the modules manifest to OrchardCore.Users to change the behaviour.

Note that by changing this view you will also effect a user editing someone elses profile, hence it works they way it does.

**Reply by jsalgado10** [reply]
@deanmarcussen thanks that worked. I added a new ``UserSaveButtons.Edit.cshtml`` view that will handle both scenarios.

```@{
    bool manageUsers = User.Claims.FirstOrDefault(c => c.Type == "Permission" && c.Value == "ManageUsers") != null;
}
<div class="form-group">
    <button class="btn btn-primary save" type="submit">@T["Save"]</button>
    @if (manageUsers)
    {
        <a class="btn btn-secondary cancel" role="button" asp-route-action="Index">@T["Cancel"]</a>
    }
    else
    {
        <a class="btn btn-secondary cancel" role="button" href="~/Admin">@T["Cancel"]</a>
    }

</div>
```

I think this could be added to the Users Module.
